msdk-io/msdk-io-netcdf/src/main/java/io/github/msdk/io/netcdf/NetCDFMsScan.java

```diff
@@ -127,15 +127,7 @@ public NetCDFMsScan(Integer scanNumber, int[] scanStartPositions, float[] scanRe
   public Integer getNumberOfDataPoints() {
     if (numOfDataPoints == null) {
       final Integer scanIndex = getScanIndex();
-      try {
-        final int scanStartPosition[] = {scanStartPositions[scanIndex]};
-        final int scanLength[] =
-            {scanStartPositions[scanIndex + 1] - scanStartPositions[scanIndex]};
-        final Array massValueArray = massValueVariable.read(scanStartPosition, scanLength);
-        numOfDataPoints = massValueArray.getShape()[0];
-      } catch (IOException | InvalidRangeException e) {
-        throw new MSDKRuntimeException(e);
-      }
+      numOfDataPoints = scanStartPositions[scanIndex + 1] - scanStartPositions[scanIndex];
```

Comment:
Isn'd this going to throw ArrayIndexOutOfBoundsException for the last scan?

Reply:
Unfortunately `scanIndex + 1 = scanNumber` only holds for Thermo instruments. For Sciex there can be rather large gaps in scan numbers.

Reply:
Do you happen to have such netcdf file?